rs/boundary_node/ic_boundary/src/core.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::disallowed_types)]
```

Comment:
[Same comment as before](https://github.com/dfinity/ic/pull/2132#discussion_r1808655207):
> Instead of allowing all disallowed types in the entire file, couldwe apply this allow-rule only to the line(s) where it actually applies? This way we would catch disallowed types in files where other disallowed types are already present. If there are multiple/too many occurrences, maybe an alias type like the one used in line 16 (use tokio::sync::Mutex as AsyncMutex;) could help.

If we leave it to the code owners to clean this up, ideally a ticket is created for them, because without a ticket this will just be forgotten once the PR is merged.

Reply:
https://dfinity.atlassian.net/browse/BOUN-1284
https://dfinity.atlassian.net/browse/CON-1430
https://dfinity.atlassian.net/browse/FI-1541